sra-tools/1.0/sra_dump.smk

```diff
@@ -0,0 +1,149 @@
+        **config["resources"]["fastq_dump"]
+    wildcard_constraints:
+        accession = "|".join(accessions["accession"].tolist())
+    group: "fasterq"
+    shell:
+        "fasterq-dump -e {threads} {params.opts} {params.ngc} {wildcards.accession}"
+
+rule gzip_fastq:
+    input:
+        fastq = "{accession}{read}.fastq"
+    output:
+        fastq = "{accession}{read}.fastq.gz"
+    threads: 1
+    resources:
+        **config["resources"]["fastq_dump"]
+    wildcard_constraints:
+        accession = "|".join(accessions["accession"].tolist())
+    group: "fasterq"
+    shell:
+        "gzip -c {input.fastq} > {output.fastq} && rm -f {input.fastq}"
+
+def get_fastq_read(wildcards):
+    # Get the path to the output directory from the checkpoint fasterq_dump
+    # **wildcards unpacks the wildcards object from the rule where this input function is called
+    checkpoint_output = checkpoints.fasterq_dump.get(accession=wildcards.accession).output.complete
+    print(glob_wildcards(checkpoint_output.replace(".fastq.complete", "") + "{read}.fastq"))
+    # Obtain the read wildcards using the glob_wildcards function
+    fastqs = expand(
+        rules.gzip_fastq.output.fastq,
+        accession=wildcards.accession,
+        read = glob_wildcards(checkpoint_output.replace(".fastq.complete", "") + "{read}.fastq").read
+        )
+    return fastqs
+
+rule fastq_complete:
+    input:
+        get_fastq_read
+    output:
+        touch("{accession}.gzip.complete")
+
+def get_genome_arg(
+    out_type = config["out_file_type"],
+    ref_fasta = config["params"]["ref"]
+):
+    if out_type == "cram":
+        assert os.path.exists(ref_fasta), (f"Reference fasta file {ref_fasta} does not exist. ")
+        param = f"-CS -T {ref_fasta}"
+    elif out_type == "bam":
+        param = "-bS"
+    else:
+        raise AssertionError(f"Specified out_file_type is \"{out_type}\", not one of [\"bam\", \"cram\"]. ")
+    return(param)
+
+
+rule sam_dump:
+    output:
+        sam = pipe("{accession}.sam")
+    params:
+        ngc = get_ngc_arg(),
+        opts = config["params"]["sam-dump"]
+    conda:
+        config["conda"]["sra-tools"]
+    threads: 1
+    resources:
+        mem_mb = 10000
+    shell:
+        "sam-dump {params.opts} {wildcards.accession} > {output.sam}"
+
+rule sam_to_bam:
+    input:
+        sam = str(rules.sam_dump.output.sam)
+    output:
+        bam = "{accession}.hg19." + config["out_file_type"],
+        complete = touch("{accession}.complete")
+    params:
+        bam_cram = get_genome_arg(),
+        opts = config["params"]["samtools"]
+    conda:
+        config["conda"]["sra-tools"]
+    threads: config["threads"]["sam_to_bam"]
+    resources:
+        **config["resources"]["sam_to_bam"]
+    shell:
+        "samtools view {params.bam_cram} -@ {threads} {input.sam} > {output.bam}"
+
+rule index_bam:
+    input:
+        bam = str(rules.sam_to_bam.output.bam)
+    output:
+        complete = touch("{accession}.indexed")
```

Comment:
I just wonder why this is not a designated index extension, because we don't know upfront if this is bam or cram?